torch/csrc/jit/fusers/cpu/fusion_compiler.cpp

```diff
@@ -14,19 +14,95 @@
 #include <sstream>
 #include <tuple>
 
+#if ((__GLIBC__ == 2) && (__GLIBC_MINOR__ >= 2)) || (__GLIBC__ > 2)
+#include <spawn.h>
+#ifdef POSIX_SPAWN_USEVFORK
+#define USE_POSIX_SPAWN
+#include <sys/wait.h>
+extern char **environ;
+#endif
+#endif
+
+#ifndef USE_POSIX_SPAWN
+#ifndef __APPLE__
+#include <malloc.h>
+#define USE_MALLOC_TRIM
+#endif
+#endif
+
 namespace torch { namespace jit { namespace cpufuser {
 
 CPUFusionCompiler& getFusionCompiler() {
   static CPUFusionCompiler compiler;
   return compiler;
 }
 
+#ifdef USE_POSIX_SPAWN
+struct PosixSpawnAttrVfork {
+  explicit PosixSpawnAttrVfork() {
+    int err = posix_spawnattr_init(&attr);
+    if (err != 0) {
+      AT_ERROR("posix_spawnattr_init: ", strerror(err));
+    }
```

Comment:
Those could get shortened to `AT_CHECK(posix_spawnattr_init(...) == 0, "some error message")`